ktx2Decoder/src/Transcoders/liteTranscoder.ts

```diff
@@ -0,0 +1,70 @@
+import { Nullable } from '../types';
+import { Transcoder, sourceTextureFormat, transcodeTarget } from '../transcoder';
+import { WASMMemoryManager } from '../wasmMemoryManager';
+import { KTX2FileReader, IKTX2_ImageDesc } from '../ktx2FileReader';
+
+/**
+ * @hidden
+ */
+export class LiteTranscoder extends Transcoder {
+
+    private _modulePath: string;
+    private _modulePromise: Promise<{ module: any }>;
+    private _memoryManager: WASMMemoryManager;
+
+    protected _loadModule(): Promise<{ module: any }> {
+        if (this._modulePromise) {
+            return this._modulePromise;
+        }
+
+        this._modulePromise = new Promise((resolve, reject) => {
+            WASMMemoryManager.LoadWASM(this._modulePath)

```

Comment:
I don't think I can because I need the result of `WebAssembly.instantiate` which is itself a Promise:
```typescript
WASMMemoryManager.LoadWASM(this._modulePath)
.then((wasmBinary) => {
    WebAssembly.instantiate(wasmBinary as ArrayBuffer, { env: { memory: this._memoryManager.wasmMemory } }).then(
        (moduleWrapper) => {
            resolve({ module: moduleWrapper.instance.exports });
        }
    );
});
```

Reply:
Did it in the last commit.